PyDSS/pyPostprocessor/PostprocessScripts/AutomatedVoltageUpgrade.py

```diff
@@ -1,1855 +1,2318 @@
+                datafile.write(line)
+        return
+
+    def reg_controls_sweep(self, upper_limit, lower_limit):
+        self.vsps = []
+        v = lower_limit * self.config["nominal_voltage"]
+        while v < upper_limit * self.config["nominal_voltage"]:
+            self.vsps.append(v)
+            v += self.config["reg_v_delta"]
+        for reg_sp in self.vsps:
+            for bandw in self.config["reg_control_bands"]:
+                dss.RegControls.First()
+                while True:
+                    regctrl_name = dss.RegControls.Name()
+                    xfmr = dss.RegControls.Transformer()
+                    dss.Circuit.SetActiveElement("Transformer.{}".format(xfmr))
+                    xfmr_buses = dss.CktElement.BusNames()
+                    xfmr_b1 = xfmr_buses[0].split(".")[0]
+                    xfmr_b2 = xfmr_buses[1].split(".")[0]
+                    # Skipping over substation LTC if it exists
+                    # for n, d in self.G.in_degree().items():
+                    #     if d == 0:
+                    #         sourcebus = n
+                    sourcebus = self.source
+                    if xfmr_b1.lower() == sourcebus.lower() or xfmr_b2.lower() == sourcebus.lower():
+                        dss.Circuit.SetActiveElement("Regcontrol.{}".format(regctrl_name))
+                        if not dss.RegControls.Next() > 0:
+                            break
+                        continue
+                    dss.Circuit.SetActiveElement("Regcontrol.{}".format(regctrl_name))
+
+                    command_string = "Edit Regcontrol.{rn} vreg={sp} band={b}".format(
+                        rn=regctrl_name,
+                        sp=reg_sp,
+                        b=bandw
+                    )
+                    dss.run_command(command_string)
+                    self.dssSolver.Solve()
+                    if not dss.RegControls.Next() > 0:
+                        break
+                self.check_voltage_violations_multi_tps(upper_limit=self.upper_limit,
+                                                        lower_limit=self.lower_limit)
+                self.reg_sweep_viols["{}_{}".format(str(reg_sp), str(bandw))] = self.severity_indices[2]
+        self.apply_best_regsetting(upper_limit=self.upper_limit)
+
+    def apply_best_regsetting(self, upper_limit):
+        print(self.reg_sweep_viols)
+        # TODO: Remove substation LTC from the settings sweep
+        best_setting = ''
+        # Start with assumption that each node has a violation at all time points and each violation if outside bounds
+        #  by upper voltage limit - basically the maximum possible severity
+        min_severity = pow(len(self.all_bus_names), 2) * len(self.config["tps_to_test"]) * upper_limit
+        for key, val in self.reg_sweep_viols.items():
+            if val < min_severity:
+                min_severity = val
+                best_setting = key
+        print(best_setting, min_severity)

```

Comment:
remove?  logger.debug?
global comment for print statements

Reply:
Removed/ modified to logger wherever present